api/src/main/java/javax/enterprise/inject/Instance.java

```diff
@@ -203,5 +203,17 @@
      *         destroying bean instances
      */
     public void destroy(T instance);
+    
+    /**
+     * <p>
+     * Instructs the container to destroy the instance. The instance passed should be a dependent scoped bean instance
```

Comment:
It's intentional. If we say `has to be dependent` it should probably fail if you pass any other instance (e.g. throw IAE). And we would like to allow users to pass anything and only destroy dependent bean instances. But I'm not native speaker so my interpretation might be wrong ;-)